### PR TITLE
fix(deps): Fix the npm audit warning for sync-exec

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Remove the package, use node's child_process.syncExec instead.

blocks #6595 

@mozilla/fxa-devs - r? It's a tad easier to review with ?w=1

### CODE REVIEW

File: tests/tools/firefox_profile.js [hunk 1]
```diff
@@ -1,31 +1,27 @@
...
+module.exports = function createProfile (config) {
+  console.log('Creating Firefox profile...');
 
-    if (profileProcess && profileProcess.status === 0) {
-      encodedProfile = profileProcess.stdout;
-    } else {
-      console.log('Note: Failed to generate a Firefox profile for this configuration.');
-    }
+  let encodedProfile = '';
+  const profileArgs = JSON.stringify(JSON.stringify(config));
```
Comment (philbooth) [inline tests/tools/firefox_profile.js]: I know it was already in the code, sorry, but I can't help noticing that this seems like an unnecessarily verbose way of writing:

```
`"${JSON.stringify(config)}"`
```

Or have I misunderstood?

Comment (shane-tomlinson) [reply tests/tools/firefox_profile.js]: > Or have I misunderstood?

No, you are correct. I'll make that update.

Comment (shane-tomlinson) [reply tests/tools/firefox_profile.js]: > No, you are correct. I'll make that update.

Actually, no, when I update to the shorter version, things blow up. It seems the intern is expecting doubly escaped input.

## using short version

```
"{"asyncTimeout":5000,"bail":false,"defaultTimeout":45000,"environments":{"browserName":"firefox"},"filterErrorStack":true,"functionalSuites":["tests/functional/email_service.js","tests/functional/mailcheck.js","tests/functional/sync_v3_email_first.js","tests/functional/fx_firstrun_v2_email_first.js","tests/functional/password_strength_experiment.js","tests/functional/sign_in.js","tests/functional/sign_in_cached.js","tests/functional/sign_in_blocked.js","tests/functional/sign_in_totp.js","tests/functional/sign_in_recovery_code.js","tests/functional/sync_sign_in.js","tests/functional/sync_force_auth.js","tests/functional/sign_up.js","tests/functional/complete_sign_in.js","tests/functional/complete_sign_up.js","tests/functional/connect_another_device.js","tests/functional/send_sms.js","tests/functional/sync_sign_up.js","tests/functional/sync_v2_sign_up.js","tests/functional/sync_v2_sign_in.js","tests/functional/sync_v2_reset_password.js","tests/functional/sync_v2_settings.js","tests/functional/sync_v2_force_auth.js","tests/functional/sync_v3_force_auth.js","tests/functional/sync_v3_reset_password.js","tests/functional/sync_v3_settings.js","tests/functional/sync_v3_sign_in.js","tests/functional/sync_v3_sign_up.js","tests/functional/fx_desktop_handshake.js","tests/functional/fx_firstrun_v1_sign_up.js","tests/functional/fx_firstrun_v1_sign_in.js","tests/functional/fx_firstrun_v1_settings.js","tests/functional/fx_firstrun_v2_sign_up.js","tests/functional/fx_firstrun_v2_settings.js","tests/functional/fx_ios_v1_sign_in.js","tests/functional/fx_ios_v1_sign_up.js","tests/functional/fx_fennec_v1_sign_in.js","tests/functional/fx_fennec_v1_force_auth.js","tests/functional/fx_fennec_v1_sign_up.js","tests/functional/fx_fennec_v1_settings.js","tests/functional/mob_android_v1.js","tests/functional/mob_ios_v1.js","tests/functional/bounced_email.js","tests/functional/legal.js","tests/functional/tos.js","tests/functional/pp.js","tests/functional/confirm.js","tests/functional/delete_account.js","tests/functional/reset_password.js","tests/functional/sync_reset_password.js","tests/functional/recovery_key.js","tests/functional/robots_txt.js","tests/functional/settings.js","tests/functional/settings_clients.js","tests/functional/settings_common.js","tests/functional/settings_change_email.js","tests/functional/settings_secondary_emails.js","tests/functional/sync_settings.js","tests/functional/change_password.js","tests/functional/force_auth.js","tests/functional/force_auth_blocked.js","tests/functional/404.js","tests/functional/500.js","tests/functional/pages.js","tests/functional/back_button_after_start.js","tests/functional/cookies_disabled.js","tests/functional/password_visibility.js","tests/functional/avatar.js","tests/functional/alternative_styles.js","tests/functional/email_opt_in.js","tests/functional/refreshes_metrics.js","tests/functional/upgrade_storage_formats.js"],"fxaAuthRoot":"http://127.0.0.1:9000/v1","fxaContentRoot":"http://127.0.0.1:3030/","fxaDevBox":false,"fxaEmailRoot":"http://127.0.0.1:9001","fxaOAuthApp":"http://127.0.0.1:8080/","fxaOAuthRoot":"http://127.0.0.1:9010","fxaProduction":false,"fxaProfileRoot":"http://127.0.0.1:1111","fxaToken":"http://","fxaTokenRoot":"http://127.0.0.1:5000/token","fxaUntrustedOauthApp":"http://127.0.0.1:10139/","pageLoadTimeout":20000,"reporters":"runner","serverPort":9090,"serverUrl":"http://127.0.0.1:9090","socketPort":9077,"tunnelOptions":{"drivers":["firefox"]},"capabilities":{"moz:firefoxOptions":{}}}"
```

## using long version

```
"{\"asyncTimeout\":5000,\"bail\":false,\"defaultTimeout\":45000,\"environments\":{\"browserName\":\"firefox\"},\"filterErrorStack\":true,\"functionalSuites\":[\"tests/functional/email_service.js\",\"tests/functional/mailcheck.js\",\"tests/functional/sync_v3_email_first.js\",\"tests/functional/fx_firstrun_v2_email_first.js\",\"tests/functional/password_strength_experiment.js\",\"tests/functional/sign_in.js\",\"tests/functional/sign_in_cached.js\",\"tests/functional/sign_in_blocked.js\",\"tests/functional/sign_in_totp.js\",\"tests/functional/sign_in_recovery_code.js\",\"tests/functional/sync_sign_in.js\",\"tests/functional/sync_force_auth.js\",\"tests/functional/sign_up.js\",\"tests/functional/complete_sign_in.js\",\"tests/functional/complete_sign_up.js\",\"tests/functional/connect_another_device.js\",\"tests/functional/send_sms.js\",\"tests/functional/sync_sign_up.js\",\"tests/functional/sync_v2_sign_up.js\",\"tests/functional/sync_v2_sign_in.js\",\"tests/functional/sync_v2_reset_password.js\",\"tests/functional/sync_v2_settings.js\",\"tests/functional/sync_v2_force_auth.js\",\"tests/functional/sync_v3_force_auth.js\",\"tests/functional/sync_v3_reset_password.js\",\"tests/functional/sync_v3_settings.js\",\"tests/functional/sync_v3_sign_in.js\",\"tests/functional/sync_v3_sign_up.js\",\"tests/functional/fx_desktop_handshake.js\",\"tests/functional/fx_firstrun_v1_sign_up.js\",\"tests/functional/fx_firstrun_v1_sign_in.js\",\"tests/functional/fx_firstrun_v1_settings.js\",\"tests/functional/fx_firstrun_v2_sign_up.js\",\"tests/functional/fx_firstrun_v2_settings.js\",\"tests/functional/fx_ios_v1_sign_in.js\",\"tests/functional/fx_ios_v1_sign_up.js\",\"tests/functional/fx_fennec_v1_sign_in.js\",\"tests/functional/fx_fennec_v1_force_auth.js\",\"tests/functional/fx_fennec_v1_sign_up.js\",\"tests/functional/fx_fennec_v1_settings.js\",\"tests/functional/mob_android_v1.js\",\"tests/functional/mob_ios_v1.js\",\"tests/functional/bounced_email.js\",\"tests/functional/legal.js\",\"tests/functional/tos.js\",\"tests/functional/pp.js\",\"tests/functional/confirm.js\",\"tests/functional/delete_account.js\",\"tests/functional/reset_password.js\",\"tests/functional/sync_reset_password.js\",\"tests/functional/recovery_key.js\",\"tests/functional/robots_txt.js\",\"tests/functional/settings.js\",\"tests/functional/settings_clients.js\",\"tests/functional/settings_common.js\",\"tests/functional/settings_change_email.js\",\"tests/functional/settings_secondary_emails.js\",\"tests/functional/sync_settings.js\",\"tests/functional/change_password.js\",\"tests/functional/force_auth.js\",\"tests/functional/force_auth_blocked.js\",\"tests/functional/404.js\",\"tests/functional/500.js\",\"tests/functional/pages.js\",\"tests/functional/back_button_after_start.js\",\"tests/functional/cookies_disabled.js\",\"tests/functional/password_visibility.js\",\"tests/functional/avatar.js\",\"tests/functional/alternative_styles.js\",\"tests/functional/email_opt_in.js\",\"tests/functional/refreshes_metrics.js\",\"tests/functional/upgrade_storage_formats.js\"],\"fxaAuthRoot\":\"http://127.0.0.1:9000/v1\",\"fxaContentRoot\":\"http://127.0.0.1:3030/\",\"fxaDevBox\":false,\"fxaEmailRoot\":\"http://127.0.0.1:9001\",\"fxaOAuthApp\":\"http://127.0.0.1:8080/\",\"fxaOAuthRoot\":\"http://127.0.0.1:9010\",\"fxaProduction\":false,\"fxaProfileRoot\":\"http://127.0.0.1:1111\",\"fxaToken\":\"http://\",\"fxaTokenRoot\":\"http://127.0.0.1:5000/token\",\"fxaUntrustedOauthApp\":\"http://127.0.0.1:10139/\",\"pageLoadTimeout\":20000,\"reporters\":\"runner\",\"serverPort\":9090,\"serverUrl\":\"http://127.0.0.1:9090\",\"socketPort\":9077,\"tunnelOptions\":{\"drivers\":[\"firefox\"]},\"capabilities\":{\"moz:firefoxOptions\":{}}}"
```

Comment (philbooth) [reply tests/tools/firefox_profile.js]: Ah, of course, yep. I guess I should have said instead:

```
`'${JSON.stringify(config)}'`
```

(but whatever, it's not important, the long version is fine too)

**Pull request merged**